src/Rules/RuleLevelHelper.php

```diff
@@ -221,23 +221,25 @@ public function findTypeToCheck(
 			return new FoundTypeResult(new ErrorType(), [], $errors, null);
 		}
 
-		if (!$this->checkUnionTypes) {
-			if ($type instanceof ObjectWithoutClassType) {
-				return new FoundTypeResult(new ErrorType(), [], [], null);
-			}
-			if ($type instanceof UnionType) {
-				$newTypes = [];
-				foreach ($type->getTypes() as $innerType) {
-					if (!$unionTypeCriteriaCallback($innerType)) {
-						continue;
-					}
-
-					$newTypes[] = $innerType;
-				}
+		if (!$this->checkUnionTypes && $type instanceof ObjectWithoutClassType) {
+			return new FoundTypeResult(new ErrorType(), [], [], null);
+		}
 
-				if (count($newTypes) > 0) {
-					return new FoundTypeResult(TypeCombinator::union(...$newTypes), $directClassNames, [], null);
+		if (
```

Comment:
While working on this issue, I got multiple error message
```
Binary operation "+" between (int|string) and int results in an error.
```
which should indeed be an error when checking benevolent union.

I discovered that in the PR https://github.com/phpstan/phpstan-src/pull/1930, I modified `accepts` but not `findTypeToCheck`.

This is fixed with this.